utils/sampler/bin/sampler.dart

```diff
@@ -78,38 +80,112 @@ String getPackageDigest(Directory packageLocation) {
   ]);
 }
 
-Future<File?> buildExecutableIfNeeded(Directory packageLocation) async {
-  final Platform platform = FlutterInformation.instance.platform;
-  final Directory flutterRoot = FlutterInformation.instance.getFlutterRoot();
-  final File mainLocation = packageLocation.childDirectory('lib').childFile('main.dart');
-  final File flutterExe =
-      flutterRoot.childDirectory('bin').childFile(platform.isWindows ? 'flutter.bat' : 'flutter');
-  final File builtExecutable = findExecutableForPlatform('sampler', packageLocation);
+Future<void> copyDirectory(Directory source, Directory destination) async {
+  await for (var entity in source.list(recursive: false)) {
+    if (entity is Directory) {
+      Directory newDirectory = destination.absolute.childDirectory(entity.basename);
+      await newDirectory.create();
+      await copyDirectory(entity.absolute, newDirectory);
+    } else if (entity is File) {
+      await entity.copy(destination.childFile(entity.basename).path);
+    }
+  }
+}
+
+void copyDirectorySync(Directory source, Directory destination) {
```

Comment:
This method uses synchronous file API (listSync, createSync, copySync), but the method above used the asynchronous versions. Why this difference?

Reply:
Oh, actually, that was just because when I was debugging, the async version was just exiting (turns out I wasn't awaiting `main`), so I made a sync version.  I'll revert to the async version.